sql/hive/src/test/scala/org/apache/spark/sql/sources/hadoopFsRelationSuites.scala

```diff
@@ -594,4 +594,29 @@ class ParquetHadoopFsRelationSuite extends HadoopFsRelationTest {
       checkAnswer(read.format("parquet").load(path), df)
     }
   }
+
+  test("SPARK-8079: Avoid NPE thrown from BaseWriterContainer.abortJob") {
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+
+      val cause = intercept[Throwable] {
+        // Parquet doesn't allow field names with spaces.  Here we are intentionally making an
+        // exception thrown from the `ParquetRelation2.prepareForWriteJob()` method to trigger
+        // the bug.  Please refer to spark-8079 for more details.
+        range(1, 10)
+          .withColumnRenamed("id", "a b")
+          .write
+          .format("parquet")
+          .save(path)
+      }
+
+      cause match {
```

Comment:
if you change the intercept to AnalysisException, you don't need the case match anymore